client/consensus/babe/src/tests.rs

```diff
@@ -435,13 +432,11 @@ fn run_one_test(mutator: impl Fn(&mut TestHeader, Stage) + Send + Sync + 'static
 				env: environ,
 				sync_oracle: DummyOracle,
 				create_inherent_data_providers: Box::new(|_, _| async {
-					let timestamp = TimestampInherentDataProvider::from_system_time();
 					let slot = InherentDataProvider::from_timestamp_and_slot_duration(
-						*timestamp,
+						sp_timestamp::current_timestamp(),
 						SlotDuration::from_millis(6000),
 					);
-
-					Ok((timestamp, slot))
+					Ok((slot,))
```

Comment:
Looks like we should also implement the trait for just `T` as well. Aka not only for tuples